src/ray/object_manager/object_manager.h

```diff
@@ -232,6 +236,59 @@ class ObjectManager : public ObjectManagerInterface {
   /// A set of active wait requests.
   std::unordered_map<UniqueID, WaitState> active_wait_requests_;
 
+  /// The cached Object transfer task due to maximum connect reached.
+  struct ObjectTransferTask {
+    ObjectID object_id;
+
+    int64_t data_size;
+
+    int64_t metadata_size;
+
+    int64_t chunk_index;
+
+    ObjectTransferTask(ObjectID object_id_, int data_size_, int64_t metadata_size_,
+                       int64_t chunk_index_)
+        : object_id(object_id_),
+          data_size(data_size_),
+          metadata_size(metadata_size_),
+          chunk_index(chunk_index_) {}
+    ObjectTransferTask() {}
+  };
+
+  template <typename T>
+  class SendingTaskPool {
+   private:
+    std::mutex mutex;
+
+    std::unordered_map<ClientID, std::list<T>> cached_tasks;
+
+   public:
+    void AddTask(const ClientID &client_id, T &&task) {
```

Comment:
don't need provide move parameter, both ObjectTransferTask and ObjectID don't provide move copy function. std::move equal copy.